contracts/liquidity_hub/pool-manager/src/contract.rs

```diff
@@ -174,9 +173,14 @@ fn optional_addr_validate(
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> Result<Binary, ContractError> {
     match msg {
-        QueryMsg::NativeTokenDecimals { denom } => Ok(to_json_binary(
-            &queries::query_native_token_decimal(deps, denom)?,
-        )?),
+        QueryMsg::NativeTokenDecimals {
+            pair_identifier,
```

Comment:
Is it really needed to use both, `pair_identifier` and `denom`? Isn't `denom` unique?

Reply:
denom is not unique no, you can have duplicated pools. So you need the pair identifier to see what the decimals are for that denom on a particular pool